### ISSUE TITLE
[Feature Request]: Add signed certificates to data connections

### ISSUE THREAD

**Issue by lucferbux** [opendatahub-io/odh-dashboard#1711]
### Solution

Mocks: https://github.com/opendatahub-io/odh-dashboard/issues/1711#issuecomment-1763941907

Likely we will just update our Data Connection value with the bundle as needed. How it interacts with each component who consumes our Data Connection values can possibly be different.

[WIP] (still getting information from each team)
* Model Serving - Deploy Models
    * Model Serving uses `AWS_CA_BUNDLE` field in our Data Connections to accept the cert ([source](https://github.com/opendatahub-io/odh-model-controller/pull/62))
* DS Pipelines - Create DSPA
    * Waiting on feedback from the backend team on what field of the DSPA points at the cert value
    * Note: we are working on [disconnecting from Data Connections](https://github.com/opendatahub-io/odh-dashboard/issues/1214), but the premise is the same
* Elyra Notebooks - Secret Update
    * Waiting on feedback from the IDE team on what field name will be in our elyra secret we provide to notebooks

---

### Feature description

We've found that Modelmesh allows to set a self-signed certificate to open a connection. But we don't support adding that field to our data connection object right now.

The Model Serving team has been working on an implementation for the modelmesh controller to allow connections with self-signing certificates https://github.com/opendatahub-io/odh-model-controller/issues/61 

There's already another spike to implement this in UX https://github.com/opendatahub-io/odh-dashboard/issues/1825

The main goal would be adding support for adding certificates through the dashboard. 

### Describe alternatives you've considered

Users could add that into the secret through Openshift.

### Anything else?

This might be tricky and we might need to have a discussion. The straightforward implementation would be adding another field to the data connection, but the issue is that it would only work for model serving, we might need to take a look.

**Comment by lucferbux** [comment]
@kywalker-rh @xianli123 @andrewballantyne @Jooho  I've opened this to track the effort in the dashboard.
I think we might need a discussion about it since it can be tricky to pull this off considering data connections are used in several components.

**Comment by xianli123** [comment]
@lucferbux @andrewballantyne I didn't find the self-signing certificates relevant resources in [the official documentation](https://access.redhat.com/documentation/en-us/red_hat_openshift_data_science/1/html/working_on_data_science_projects/working-on-data-science-projects_nb-server#using_data_connections). Could you please share some resources of the self-signing certificates? I want to learn more knowledge about it. Thanks!

**Comment by Jooho** [comment]
@xianli123 
modelmesh, kserve pull a model from S3 or minio and so on. Some registries use their own certificate that was signed by their own root ca. In order to pull a model from those registries, they have to provide a custom cert to access the registries. Modelmesh/kserver can set this cert in Secret object. 

Using dashboard, user uses data-connection object to specify storage information and odh-model-controller converts the information into a Secret object. The Secret object is managed by operator so it is not possible to customize. 

For this reason, I tried to use data connection but it might affect to other components.
I hope this help you understand better.

**Comment by xianli123** [comment]
@Jooho Thanks. I can roughly understand the use cases of that. It's very helpful.

**Comment by andrewballantyne** [comment]
Design effort will be tracked by https://github.com/opendatahub-io/odh-dashboard/issues/1825

**Comment by xianli123** [comment]
Hi @Jooho I am wondering what attributes will be included in the signed certificates. Let me take the source secret as an example as shown below. The secret includes `Authentication type` and `Private key`. What attributes do the signed certificates include? Thanks!

![image](https://github.com/opendatahub-io/odh-dashboard/assets/53931519/0fd19076-ff48-420a-9393-cbfb864c84d7)

**Comment by xianli123** [comment]
Hi @Jooho @lucferbux @andrewballantyne I came up with a design for the self-signed cert as shown below. I just added an optional field named `CA bundle` that allows users to upload or type the bundle. Does it make sense?
<img width="574" alt="Screenshot 2023-10-16 at 16 03 06" src="https://github.com/opendatahub-io/odh-dashboard/assets/53931519/dd7ef165-46b8-442a-aae2-987e37b4bdd3">

**Comment by andrewballantyne** [comment]
Looking good @xianli123 -- for Model Serving we need to talk how you guys are going to get access to this @Jooho

@HumairAK we also need to talk about how this can apply to DS Pipelines as well. Do you guys have a path forward to allowing us to hand off a cert value like this?

@harshad16 we'll probably need to pass this off to Elyra too.

**Comment by xianli123** [comment]
@andrewballantyne Thanks for reviewing the solution. I noticed that you added some new items related to the self-signed certs and have looked through the Slack thread. Once all the items are ready, I will start to do further UX work.

**Comment by andrewballantyne** [comment]
Understood. I think we need to have an architectural solution in place for the cluster -- I do not see it completely removing this solution you have designed. So this is just on hold until we determine the larger case.

**Comment by MatejHorinek** [comment]
As well tracked in https://issues.redhat.com/browse/RHOAIENG-108

**Comment by andrewballantyne** [comment]
Migrated to https://issues.redhat.com/browse/RHOAIENG-1813, in part with other issues. See the Jira for more linkage/relationship to other efforts.
